cmd/pdatagen/internal/base_structs.go

```diff
@@ -308,6 +308,9 @@ func (ms ${structName}) InitEmpty() {
 //
 // Important: All other functions will cause a runtime error if this returns "true".
 func (ms ${structName}) IsNil() bool {
+	if ms.orig == nil {
```

Comment:
This is not a valid state for the object. We have a comment specifically that addresses this:

>// Important: zero-initialized instance is not valid for use.

When did you encounter this?

Reply:
> // Important: zero-initialized instance is not valid for use.

How does the consumer know that the instance it got is zero-initialized? There should be an API to check for that.

I am working on feature in Jaeger where I had to check for the nil and I was getting panics.

Reply:
I think this should never happen, if everyone uses the API correctly you cannot get an instance that is not created with NewXYZ